subsys/bluetooth/host/smp.c

```diff
@@ -4903,6 +4913,18 @@ int bt_smp_auth_passkey_confirm(struct bt_conn *conn)
 	return 0;
 }
 
+int bt_smp_le_oob_set_tk(u8_t *tk)
```

Comment:
I have mixed feeling about this kind of global switch API,  shouldn't this be configurable per remote?
Also would be nice to have const pointers where possible so that one can pass const data without casting.

Reply:
In this case I was following the same pattern as for legacy passkey and the `bt_passkey_set()` function. Alternatively, it could be added to `struct bt_smp` similarly to how SC OOB data is set

Reply:
@sjanc I propose then a different API:
```
int bt_le_oob_set_legacy_tk(struct bt_conn *conn, const u8_t *tk);
```

That would be backed by adding the following to `smp.c`
```
/* SMP channel specific context */
struct bt_smp {
        ...
        /* Legacy OOB tk */
        u8_t                            oob_legacy_tk[16];
        ...
}
```

Reply:
@ivan98ams Sorry for the delay.

You copied the use case of `int bt_passkey_set(unsigned int passkey);` which is used for Fixed passkey. This is not what we want.

I would prefer instead that we had a callback similar to `void (*passkey_entry)(struct bt_conn *conn);` and it's matching `int bt_conn_auth_passkey_entry(struct bt_conn *conn, unsigned int passkey);`

For LE SC with OOB the similar handling is with 
`	void (*oob_data_request)(struct bt_conn *conn, struct bt_conn_oob_info *info);`
And it's expected function to be called:
In fact oob_data_request can already be used. Setting `bt_conn_oob_info.type = BT_CONN_OOB_LE_LEGACY`

bt_le_oob_set_sc_data